### DISCUSSION TITLE
Table BulkAction doesn`t work - changes are rolled back from database

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#2144] [Help]
On my table in resource page I got bulk action, which after calling doesn`t change data as expected.

Here is my BulkAction:

```php
->bulkActions([
  BulkAction::make('toggle_hidden')
    ->label('Show/ hide')
    ->action(fn (Collection $records) => $records->each->toggleHidden())
    ->deselectRecordsAfterCompletion(),
]);
```

and this is **toggleHidden()** method from my Model:

```php
public function toggleHidden()
{
  $this->is_hidden = !$this->is_hidden;
  $this->save();
}
```

When I dump value after beeing saved in my Model, it shows that value has changed as expected, so something is happening after that. 

Console doesnt show any errors. It doesnt look like there is any extra request. I simplified for test Model to only this method and relation definitions. Similiary about table - it got only one column. I checked also that very similiar BulkAction like this works as intended:

```php
BulkAction::make('delete')
  ->label('Usuń')
  ->action(fn (Collection $records) => $records->each->delete())
  ->deselectRecordsAfterCompletion(),
```

Any ideas what can cause this issue?

**Comment by danharrin** [comment]
If you refresh the page after running the action, are the changes there?

**Reply by studiowizjo** [reply]
No. It is the same as before using bulk action.

**Reply by danharrin** [reply]
I'm honestly not sure what to suggest, can you see which database query is wrong with debugbar?

**Comment by studiowizjo** [comment]
It is second update query which reverse change, but not idea from where it comes from (?)

![screencapture-localhost-8000-admin-invoices-2022-04-14-19_33_53](https://user-images.githubusercontent.com/3818874/163443706-fd967d7e-bbd5-47f9-bba9-c2fd81500b1f.jpg)

**Reply by danharrin** [reply]
Is the bulk action being executed twice?

**Reply by studiowizjo** [reply]
I clicked it only once, but dont know if internally it is executed more than once. It seems it is, because query stack shows only queries from Filament.

It is not the point now, but it is strange that same queries are doubled or tripled in a row.

**Reply by danharrin** [reply]
If you'd like to submit an issue with reproducible code, go ahead

**Reply by studiowizjo** [reply]
I finally found out an issue. It was because published table views which were out of date.

**Reply by danharrin** [reply]
Please remove them if you're not doing anything :)

Use CSS for customization where you can, there's many `filament-*` CSS properties you can use.
